app/actions/remote/entry/common.ts

```diff
@@ -191,14 +194,49 @@ const fetchAppEntryData = async (serverUrl: string, sinceArg: number, initialTea
     // Fetch in parallel teams / team membership / channels for current team / user preferences / user
     const promises: [Promise<MyTeamsRequest>, Promise<MyChannelsRequest | undefined>, Promise<MyUserRequest>] = [
         fetchMyTeams(serverUrl, fetchOnly),
-        initialTeamId ? fetchMyChannelsForTeam(serverUrl, initialTeamId, includeDeletedChannels, since, fetchOnly, false, isCRTEnabled) : Promise.resolve(undefined),
+        onLoadTeamId ? fetchMyChannelsForTeam(serverUrl, onLoadTeamId, includeDeletedChannels, since, fetchOnly, false, isCRTEnabled) : Promise.resolve(undefined),
         fetchMe(serverUrl, fetchOnly),
     ];
 
     const resolution = await Promise.all(promises);
     const [teamData, , meData] = resolution;
     let [, chData] = resolution;
 
+    let initialTeamId = onLoadTeamId;
+    let initialChannelId = channelId;
+    let gmConverted = false;
+
+    if (channelId && chData?.channels) {
```

Comment:
I wonder if this whole block should be deffered to the end of the entry logic

Also, have we done something similar for when the websocket reconnects?

Reply:
> I wonder if this whole block should be deffered to the end of the entry logic

We want to do this as early in code as possible, because in this if block we're changing channel and team ID, which are later being used in many more places in same function.

> Also, have we done something similar for when the websocket reconnects?

So the function I've handled this is in called for both startup and websocket reconnect.